public/src/main/java/com/revenuecat/purchases/models/StoreProduct.kt

```diff
@@ -43,7 +43,7 @@ interface StoreProduct : Parcelable {
     val subscriptionPeriod: String?
 
     /**
-     * List of PurchaseOptions.
+     * List of PurchaseOptions. Empty list for INAPP products.
```

Comment:
Hmm for the purchase implementation in `PurchaseWrapper` I was assuming there would always be at least 1 `PurchaseOption`, even for one time products... But then again I only checked for BC4 and Amazon [at the time](https://revenuecat.slack.com/archives/C03M70X8DGF/p1666956760783019) and the `PurchaseOption` abstraction doesn't make much sense for one-time products... I will modify my code so it works for one-time products as well!

Reply:
Why wouldn't it make sense for the OTP? I think I would keep the PurchaseOption to have a clean/common interface.
What I'm starting to question a bit more is the baseplan vs. offers separation - it seems a bit artifical to me the more I look at it.

Reply:
Well `PurchaseOption` currently contains a list of `PricingPhase`, `tags` and a `token`. None of the 3 applies to OTP I think?  I mean, we can have a single `PricingPhase` if we want to have the same abstraction but we don't really need it if we have `oneTimeProductPrice` in the `StoreProduct`. I think we either:
- Make the abstraction common between OTP and subscriptions. In this case, I think we should:
  - Remove the `oneTimeProductPrice` from the `StoreProduct`
  - OTP should also have a single `PurchaseOption` with a single `PricingPhase`.
  - This would make it slightly harder to access prices for OTP but would have a consistent interface with devs
- Make OTP not have `PurchaseOptions` and be purchased through the `GoogleStoreProduct` directly.
  - This would be slightly easier for users that use OTP to access price info.

Personally it feels to me that OTP don't need the `PurchaseOption` abstraction but I understand if we want to keep it common to avoid increasing the size of the API.


About the baseplan vs rest of the offers separation. I think we currently only need the base plan internally to be able to find a "best offer" for our deprecated APIs/hybrids. Not sure if we need the differentiation anywhere else?

Reply:
Yeah, you're right that the PurchaseOption with pricingphases doesn't apply to the OTP. It might have to be even more generic for that and then distringuish between a recurrent/one-time purchase.
Which would be good to model anyways... 
I don't want to block this PR in any way, so we can keep discussing.

Re: choosing the best offer. I don't think you need to know it's a baseplan necessarily (I think any PO with more than 1 pricing phase is likely to be better than one with a single phase)

Reply:
I agree that it feels that an OTP shouldn't have options, nor pricing phases. At the end of the day it's a one time thing and not a phased payment.